### PR TITLE
Accept broad SAML validity windows (SIRI-1201)

### PR DESCRIPTION
**Pull request by jakobvogel**
### Description

This PR fixes SAML responses from IdPs that issue broad assertion condition windows, such as ADFS responses with several hours between `NotBefore` and `NotOnOrAfter`.

Instead of rejecting responses because the IdP condition window is broad, SAML responses are accepted when they are currently valid according to `IssueInstant`, optional `NotBefore`, and the effective earliest `NotOnOrAfter`. To keep stale responses and Redis replay retention bounded, the effective acceptance deadline is capped by a configurable duration after `Assertion@IssueInstant`, with an absolute maximum of 30 minutes.

Redis-backed replay protection reserves response/assertion IDs until the earlier of the SAML `NotOnOrAfter` deadline and the local acceptance deadline, plus clock skew. This keeps duplicate submissions blocked without allowing very distant IdP expiry timestamps to create long-lived replay keys.

The SAML tests now document the relevant edge cases: optional `NotBefore`, independent `NotOnOrAfter` sources, malformed timestamps, duplicate `Conditions`, earliest deadline selection, stale response rejection, config parsing for the acceptance duration, and replay behavior for absent IDs.

### Additional Notes

- This PR fixes or works on following ticket(s): [SIRI-1201](https://scireum.myjetbrains.com/youtrack/issue/SIRI-1201)

### Checklist

- [x] Code change has been tested and works locally
- [x] Code was formatted via IntelliJ and follows SonarLint & [best practices](https://scireum.myjetbrains.com/youtrack/articles/MISC-A-16/CodeStyle-JavaDoc)

### Test Plan

- `mvn -Dtest=sirius.biz.saml.SamlTest test`

### CODE REVIEW

File: src/main/java/sirius/biz/saml/SamlHelper.java [hunk 1]
```diff
@@ -373,35 +376,26 @@ private Element selectSingleElement(Document document, @Nullable String namespac
...
-        Instant validFrom = notBefore.orElse(issueInstant);
-        if (Duration.between(validFrom, notOnOrAfter).compareTo(MAX_ASSERTION_VALIDITY) > 0) {
-            throw Exceptions.createHandled()
-                            .withSystemErrorMessage(
-                                    "Invalid SAML Response: Assertion validity exceeds maximum duration of %s.",
-                                    MAX_ASSERTION_VALIDITY)
-                            .handle();
-        }
-
         Instant deadline = notOnOrAfter.plus(SAML_CLOCK_SKEW);
```
[the remark this answers is not in the document]
Comment (jakobvogel) [reply src/main/java/sirius/biz/saml/SamlHelper.java]: done, added an internal deadline

**Review by sabieber — APPROVED** [review]
